### PR TITLE
set ntp config module

### PR DESCRIPTION
**Pull request by joechainz**

### CODE REVIEW

**Review by waynew — CHANGES_REQUESTED** [review]
Looks excellent! The only thing that I think we could do here is write a quick unit tests:

```
def test_whatever(fake_service_instance):
    esxi.set_ntp_config(service_instance=fake_service_instance)  # and whatever else
```

May or may not be worth adding some other bits :shrug:. That's just to make sure the service instance bit works as expected :shrug:

File: src/saltext/vmware/modules/esxi.py [hunk 1]
```diff
@@ -1316,6 +1316,75 @@ def get_ntp_config(
...
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+
+    try:
+        for host in hosts:
+            date_time_manager = host.configManager.dateTimeSystem
+
+            ntp_config = vim.host.NtpConfig()
```
[the remark this answers is not in the document]
Comment (joechainz) [reply src/saltext/vmware/modules/esxi.py:1375]: pyvmomi library

File: src/saltext/vmware/modules/esxi.py [hunk 2]
```diff
@@ -1316,6 +1316,75 @@ def get_ntp_config(
         raise salt.exceptions.SaltException(str(exc))
 
 
+def set_ntp_config(
```
[the remark this answers is not in the document]
Comment (waynew) [reply src/saltext/vmware/modules/esxi.py:1319]: Were those just for connecting? Our `get_service_instance` will read those values from the config.

File: src/saltext/vmware/modules/esxi.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (joechainz) [reply src/saltext/vmware/modules/esxi.py:1375]: This comes from the pyvmomi lib.

**Review by dmurphy18 — APPROVED** [review]

**Review by waynew — APPROVED** [review]
looks good, I'm down to merge tomorrow unless there are objections 👍

**Review by waynew — CHANGES_REQUESTED** [review]
Oh yeah - @dmurphy18 made a great point, we need a changelog file here

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,3 +7,7 @@ This project uses [CalVer][calver] - YY.M.D.MICRO
 - NSX functionality
 - VMC functionality
 - ESXi functionality
+
+# [22.11.18]
```
Comment (waynew) [inline CHANGELOG.md:11]: **suggestion** this should be in the changelog/NNN.added - if there was no issue created here in the repo can just be 323.added (and it should say "Added vmware_esxi.set_ntp_config module" probably)

**Review by waynew — CHANGES_REQUESTED** [review]

**Review by waynew — APPROVED** [review]
:+1: LGTM

**Comment by waynew** [comment]
Pre-commit just failing on the changelog file, we should probably see if we can make that... not happen :joy:

**Pull request merged**